daemon/src/daemon.rs

```diff
@@ -830,24 +837,111 @@ impl Daemon {
+
+                                    document_handle
+                                        .send_message(DocMessage::UpdateFile {
+                                            file_path: path
+                                                .to_str()
+                                                .expect("Failed to convert path to string")
+                                                .into(),
+                                            content,
+                                        })
+                                        .await;
+                                }
+                            }
+                        }
+                        notify::event::EventKind::Modify(notify::event::ModifyKind::Name(
+                            notify::event::RenameMode::Both,
+                        )) => {
+                            let from_path = &event.paths[0];
+                            let to_path = &event.paths[1];
+
+                            // TODO: Avoid repeating the effects here.
+                            // TODO: Is there a cleverer way to do renames in the CRDT?
+
                             document_handle
                                 .send_message(DocMessage::RemoveFile {
-                                    file_path: path
+                                    file_path: from_path
                                         .to_str()
                                         .expect("Failed to convert path to string")
                                         .into(),
                                 })
                                 .await;
+
+                            if !sandbox::ignored(&base_dir_cloned, to_path)
+                                .expect("Could not determine ignored status of file")
+                            {
+                                let bytes = sandbox::read_file(&base_dir_cloned, to_path)
+                                    .expect("Failed to read created file");
+                                let content =
+                                    String::from_utf8(bytes).expect("Could not read file as UTF-8");
+
+                                document_handle
+                                    .send_message(DocMessage::CreateFile {
+                                        file_path: to_path
+                                            .to_str()
+                                            .expect("Failed to convert path to string")
+                                            .into(),
+                                        content,
+                                    })
+                                    .await;
+                            }
+                        }
+                        e => {
+                            // Don't handle other events.
+                            // But log them! I'm curious what they are!
+                            info!("{:?}: {e:?}", event.paths);
```

Comment:
I think this is a bit too noisy for actual use. Maybe put it to `debug!`?

Reply:
Or yeah, we can also just drop them at this point. The remaining once I see are Access-type events that seem to happen after each write.

Reply:
Well, it's still interesting for now. I do see for example a couple of different `Modify(...)` events that were not matched above. See also my comment above.